sklearn/calibration.py

```diff
@@ -753,3 +755,208 @@ def calibration_curve(y_true, y_prob, *, normalize=False, n_bins=5,
+
+        ax.set(xlabel="Mean predicted probability",
+               ylabel="Fraction of positives")
+
+        self.ax_ = ax
+        self.figure_ = ax.figure
+        return self
+
+
+def plot_calibration_curve(estimator, X, y, *,
+                           n_bins=5, strategy='uniform',
+                           name=None, ref_line=True, ax=None, **kwargs):
+    """Plot calibration curve, also known as reliability diagrams, for binary
+    classifiers.
+
+    The average predicted probability for each bin is plotted on the x-axis
+    and the fraction of positive classes in each bin is plotted on the y-axis.
+
+    Extra keyword arguments will be passed to :func:`matplotlib.pyplot.plot`.
+
+    Read more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    ----------
+    estimator : estimator instance
+        Fitted classifier or a fitted :class:`~sklearn.pipeline.Pipeline` in
+        which the last estimator is a classifier. The classifier must
+        have a :term:`predict_proba` method.
+
+    X : {array-like, sparse matrix} of shape (n_samples, n_features)
+        Input values.
+
+    y : array-like of shape (n_samples,)
+        Binary target values.
+
+    n_bins : int, default=5
+        Number of bins to discretize the [0, 1] interval into when calculating
+        the calibration curve. A bigger number requires more data.
+
+    strategy : {'uniform', 'quantile'}, default='uniform'
+        Strategy used to define the widths of the bins.
+
+        - `'uniform'`: The bins have identical widths.
+        - `'quantile'`: The bins have the same number of samples and depend on
+          predicted probabilities.
+
+    name : str, default=None
+        Name for labeling curve. If `None`, the name of the estimator is used.
+
+    ref_line : bool, default=True
+        If `True`, plots a reference line representing a perfectly calibrated
+        classifier.
+
+    ax : matplotlib axes, default=None
+        Axes object to plot on. If `None`, a new figure and axes is created.
+
+    **kwargs : dict
+        Keyword arguments to be passed to :func:`matplotlib.pyplot.plot`.
+
+    Returns
+    -------
+    display : :class:`~sklearn.calibration.CalibrationDisplay`.
+        Object that stores computed values.
+
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt                     # doctest: +SKIP
+    >>> from sklearn import (datasets, calibration, model_selection,
+    ...                      linear_model)
+    >>> X, y = datasets.make_classification(random_state=0)
+    >>> X_train, X_test, y_train, y_test = model_selection.train_test_split(
+    ...     X, y, random_state=0)
+    >>> clf = linear_model.LogisticRegression(random_state=0)
+    >>> clf.fit(X_train, y_train)
+    LogisticRegression(random_state=0)
+    >>> metrics.plot_calibration_curve(clf, X_test, y_test)  # doctest: +SKIP
+    >>> plt.show()                                           # doctest: +SKIP
```

Comment:
this fails: metrics isn't imported and `plot_calibration_curbe` isn't there anyway. We need to be careful about skipping doctests ;)